### PR TITLE
[react-slick] update typings

### PR DESCRIPTION
**Pull request by r3nya**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/akiran/react-slick/blob/master/src/arrows.js#L21
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Issue: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/21473

### CODE REVIEW

File: types/react-slick/index.d.ts [hunk 1]
```diff
@@ -9,9 +11,13 @@ import * as React from "react";
 type ComponentConstructor<TProps> = React.ComponentClass<TProps> | React.StatelessComponent<TProps>;
 
 export interface CustomArrowProps {
+  key?: string | number;
```
Comment (GiedriusGrabauskas) [inline types/react-slick/index.d.ts]: `key` already exists in React component.

File: types/react-slick/index.d.ts [hunk 2]
```diff
@@ -9,9 +11,13 @@ import * as React from "react";
...
+  "data-role"?: string;
```
Comment (GiedriusGrabauskas) [inline types/react-slick/index.d.ts]: This prop doesn't have an implementation and is unchangeable. So, I think it shouldn't be here.

**Review by GiedriusGrabauskas — CHANGES_REQUESTED** [review]

File: types/react-slick/index.d.ts (hunk 1 above)
Comment (r3nya) [reply types/react-slick/index.d.ts]: Ah, right. Sorry … Will fix it.

File: types/react-slick/index.d.ts (hunk 2 above)
Comment (r3nya) [reply types/react-slick/index.d.ts]: Ok!

File: types/react-slick/index.d.ts (hunk 1 above)
Comment (r3nya) [reply types/react-slick/index.d.ts]: Fixed it. Please review.

File: types/react-slick/index.d.ts (hunk 2 above)
Comment (r3nya) [reply types/react-slick/index.d.ts]: Fixed it. Please review.

**Review by GiedriusGrabauskas — APPROVED** [review]
Thanks 👍

**Pull request merged**